### DISCUSSION TITLE
Wildcards in topic names

### DISCUSSION THREAD

**Discussion by StephenR1** [theengs/gateway#88] [Q&A]
Hi,

Is there a way to disable using wildcards in the status topic properties?
I am trying to get TheengsGateway to connect to Domoticz via the homeassistant autodiscovery feature and I think the problem I have is down to the use of wildcards.
I think it looks like domoticz is subscribing to the wildcard topic (and is receiving the messages fine via MQTT) but the messages come back with the full topic name rather than the wildcarded version and the Domoticz code is using a match on the topic string to link the internal sensor properties with the incoming message.

The alternative is adding a new bit of code to do wildcard matching in Domoticz but if it is easy in some way to disable it in TheengsGateway then that might be an easier route!

Fantastic app by the way!

Thanks

Stephen

**Comment by 1technophile** [comment]
Hi,

> Is there a way to disable using wildcards in the status topic properties?

Not for the moment, as these wildcards are hard coded.

They enable to setup multigateways installation with several Theengs Gateway and/or OpenMQTTGateway.
So that one sensor can be fed by multiple gateways.

For example:
```json
{"stat_t":"+/+/BTtoMQTT/112233445566","dev_cla":"temperature","unit_of_meas":"°C","name":"IBT-4X(S/C)-tempc","uniq_id":"112233445566-tempc","val_tpl":"{{ value_json.tempc | is_defined }}","state_class":"measurement","device":{"identifiers":["112233445566"],"connections":[["mac","10082C40ABE6"]],"manufacturer":"Inkbird","model":"IBT-4X(S/C)","name":"iBBQ-40ABE6","via_device":"OMG"}}
```

For example this sensor entity definition can be fed by either instance of OpenMQTTGateway or Theengs Gateway, even if each republish a discovery topic.

If we replace the wildcard by `home/<gateway_name1>`  the issue is that the sensor will look only at one gateway (gateway_name1) data at a time and don't retrieve the value from the others. So it will work on a single gateway installation but not on a multi gateway installation.

**Reply by StephenR1** [reply]
Thanks very much 1technophile.

I saw that the wildcard topics were put in to allow that (in 2020?) and totally see/understand that it is useful to have. Also absolutely that it would be the default.
I guess this was only a hope that there might be a flag that could override it where wildcards caused a problem.
I am hoping to write a quick patch for Domoticz (which I think is a better way to fix) and will offer it to them to use if they think it would be more generally useful.

Thanks again for the library.

Stephen
